src/internal/document_api/mod.rs

```diff
@@ -957,15 +957,17 @@ pub async fn decrypt_document<CR: rand::CryptoRng + rand::RngCore>(
         &device_private_key.recrypt_key(),
     )?;
 
-    aes::decrypt(&mut enc_doc, *sym_key.bytes())
-        .map_err(|e| e.into())
-        .map(move |decrypted_doc| DocumentDecryptResult {
-            id: doc_meta.0.id,
-            name: doc_meta.0.name,
-            created: doc_meta.0.created,
-            updated: doc_meta.0.updated,
-            decrypted_data: decrypted_doc.to_vec(),
-        })
+    Ok(
+        aes::decrypt(&mut enc_doc, *sym_key.bytes()).map(move |decrypted_doc| {
+            DocumentDecryptResult {
+                id: doc_meta.0.id,
+                name: doc_meta.0.name,
+                created: doc_meta.0.created,
+                updated: doc_meta.0.updated,
+                decrypted_data: decrypted_doc.to_vec(),
+            }
+        })?,
```

Comment:
Just curious, is this a more Rust-preferred way of doing this?

Reply:
I would say yes, so long as you are just calling `into()` on an error and not inspecting the error at all.